libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Rotoye.cpp

```diff
@@ -0,0 +1,162 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_BattMonitor.h"
+#include "AP_BattMonitor_SMBus_Rotoye.h"
+#include <utility>
+
+#define BATTMONITOR_SMBUS_Rotoye_NUM_CELLS 10
+uint8_t Rotoye_cell_ids[] = { 0x3f,  // cell 1
+                              0x3e,  // cell 2
+                              0x3d,  // cell 3
+                              0x3c,  // cell 4
+                              0x3b,  // cell 5
+                              0x3a,  // cell 6
+                              0x39,  // cell 7
+                              0x38,  // cell 8
+                              0x37,  // cell 9
+                              0x36}; // cell 10
+
+#define SMBUS_READ_BLOCK_MAXIMUM_TRANSFER    0x20   // A Block Read or Write is allowed to transfer a maximum of 32 data bytes.
+
+/*
+ * Other potentially useful registers, listed here for future use
+ * #define BATTMONITOR_SMBUS_Rotoye_CHARGE_STATUS         0x0d    // relative state of charge
+ * #define BATTMONITOR_SMBUS_Rotoye_BATTERY_STATUS        0x16    // battery status register including alarms
+ * #define BATTMONITOR_SMBUS_Rotoye_BATTERY_CYCLE_COUNT   0x17    // cycle count
+ * #define BATTMONITOR_SMBUS_Rotoye_DESIGN_VOLTAGE        0x19    // design voltage register
+ * #define BATTMONITOR_SMBUS_Rotoye_MANUFACTURE_DATE      0x1b    // manufacturer date
+ * #define BATTMONITOR_SMBUS_Rotoye_SERIALNUM             0x1c    // serial number register
+ * #define BATTMONITOR_SMBUS_Rotoye_HEALTH_STATUS         0x4f    // state of health
+ * #define BATTMONITOR_SMBUS_Rotoye_SAFETY_ALERT          0x50    // safety alert
+ * #define BATTMONITOR_SMBUS_Rotoye_SAFETY_STATUS         0x51    // safety status
+ * #define BATTMONITOR_SMBUS_Rotoye_PF_ALERT              0x52    // safety status
+ * #define BATTMONITOR_SMBUS_Rotoye_PF_STATUS             0x53    // safety status
+*/
+
+// Constructor
+AP_BattMonitor_SMBus_Rotoye::AP_BattMonitor_SMBus_Rotoye(AP_BattMonitor &mon,
+                                                   AP_BattMonitor::BattMonitor_State &mon_state,
+                                                   AP_BattMonitor_Params &params,
+                                                   AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
+    : AP_BattMonitor_SMBus(mon, mon_state, params, std::move(dev))
+{}
+
+void AP_BattMonitor_SMBus_Rotoye::timer()
+{
+	// check if PEC is supported
+    if (!check_pec_support()) {
+        return;
+    }
+
+    uint16_t data;
+    uint32_t tnow = AP_HAL::micros();
```

Comment:
const?

Reply:
This bit of code is the same as in the AP_Battmonitor_SMBus_Maxell driver which I used for reference

Reply:
Ideally it should be const just to improve it, but it's not a big deal.